### PR TITLE
Clear IP bgp, show peer group support and bug fixes

### PR DESCRIPTION
**Pull request by venkatmahalingam**

### CODE REVIEW

File: models/yang/sonic/sonic-bgp-clear.yang [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        leaf clear-all {
+            type boolean;
+            description
+                "Clear all BGP peers.";
+        }
+
+        leaf vrf-name {
+            type string;
+            description
+                "Clear BGP by VRF name.";
```
Comment (venkatmahalingam) [inline models/yang/sonic/sonic-bgp-clear.yang:62]: default VRF name as default value?

File: models/yang/sonic/sonic-bgp-clear.yang [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        }
+
+        leaf family {
+            type enumeration {
+                enum IPv4;
+                enum IPv6;
+            }
```
Comment (venkatmahalingam) [inline models/yang/sonic/sonic-bgp-clear.yang:69]: IPv4 family as default value

File: models/yang/sonic/sonic-bgp-clear.yang [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                    }
+                    type leafref {
+                        path "/svlan:sonic-vlan/svlan:VLAN/svlan:VLAN_LIST/svlan:name";
+                    }
+                    type leafref {
+                        path "/smt:sonic-mgmt-port/smt:MGMT_PORT/smt:MGMT_PORT_LIST/smt:ifname";
+                    }
+                }
+                description
+                    "Interface name: Parameter for deleting all ARP  entries
```
Comment (venkatmahalingam) [inline models/yang/sonic/sonic-bgp-clear.yang]: Not ARP entries.. please fix description.

File: models/yang/sonic/sonic-bgp-clear.yang [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                        path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+                    }
+                    type leafref {
+                        path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                    }
+                    type leafref {
+                        path "/svlan:sonic-vlan/svlan:VLAN/svlan:VLAN_LIST/svlan:name";
+                    }
+                    type leafref {
+                        path "/smt:sonic-mgmt-port/smt:MGMT_PORT/smt:MGMT_PORT_LIST/smt:ifname";
```
Comment (venkatmahalingam) [inline models/yang/sonic/sonic-bgp-clear.yang]: I dont think, we have any usecase with mgmt-intf, please remove it.

File: models/yang/annotations/sonic-bgp-clear-annot.yang [hunk 5]
```diff
@@ -0,0 +1,16 @@
+module sonic-bgp-clear-annot {
+
+    yang-version "1";
```
Comment (venkatmahalingam) [inline models/yang/annotations/sonic-bgp-clear-annot.yang]: 1.1?

File: src/CLI/actioner/sonic_cli_clear_bgp.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/python
+###########################################################################
+#
+# Copyright 2019 Dell, Inc.
```
Comment (venkatmahalingam) [inline src/CLI/actioner/sonic_cli_clear_bgp.py]: 2020?

File: src/translib/transformer/xfmr_bgp.go [hunk 7]
```diff
@@ -588,3 +608,174 @@ var DbToYang_bgp_gbl_afi_safi_addr_key_xfmr KeyXfmrDbToYang = func(inParams Xfmr
...
+            cmd = cmd + in
+        }
+
+        if out != "" {
+            cmd = cmd + out
+        }
+
+    }
+    cmd = strings.TrimSuffix(cmd, " ")
+    exec_vtysh_cmd (cmd)
```
Comment (venkatmahalingam) [inline src/translib/transformer/xfmr_bgp.go:777]: can you handle the error and fill status accordingly?

Comment (AshokDaparthi) [reply src/translib/transformer/xfmr_bgp.go:777]: clear is not returning proper json for FRR

File: models/yang/sonic/sonic-bgp-clear.yang (hunk 3 above)
Comment (AshokDaparthi) [reply models/yang/sonic/sonic-bgp-clear.yang]: Done

File: models/yang/sonic/sonic-bgp-clear.yang (hunk 1 above)
Comment (AshokDaparthi) [reply models/yang/sonic/sonic-bgp-clear.yang:62]: Done

File: models/yang/annotations/sonic-bgp-clear-annot.yang [hunk 8]
```diff
@@ -0,0 +1,16 @@
+module sonic-bgp-clear-annot {
+
+    yang-version "1.1";
+
+    namespace "http://openconfig.net/yang/annotation/sbgpclear-annot";
```
Comment (mhampasagar) [inline models/yang/annotations/sonic-bgp-clear-annot.yang]: Can u check the namespace , I think sonic YANG it has to Azure right?
"http://github.com/Azure/"

File: src/CLI/actioner/sonic_cli_clear_bgp.py [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+import sys
+import time
+import json
+import ast
+import netaddr
+from rpipe_utils import pipestr
+import cli_client as cc
+from scripts.render_cli import show_cli_output
+from bgp_openconfig_to_restconf_map import restconf_map
```
Comment (mhampasagar) [inline src/CLI/actioner/sonic_cli_clear_bgp.py]: restconf_map looks like unused, pl remove it.
check other import to make sure, no unused imports

**Review by AshokDaparthi — APPROVED** [review]

**Pull request merged**